### ISSUE TITLE
podman --root /tmp1/mycontainers load -i localimage.tar is not working

### ISSUE THREAD

**Issue by holgrrr** [containers/podman#19402]
### Issue Description

TL;DR:
- additionalimagestore:  /tmp1/imagestore
- tar-image stored locally /tmp1/localimage.tar
- podman --root /tmp1/mycontainers load -i localimage.tar  is not working.
 

Long version with more info:

```
Configuration and preconditions
[user@workstation /tmp1]$ cat ~/.config/containers/containers.conf 
[containers]
default_sysctls = []
 
[engine]
runtime = "/usr/bin/crun"
 
 
 
[user@workstation /tmp1]$ cat ~/.config/containers/storage.conf 
[storage]
driver = "overlay"
rootless_storage_path = "/tmp1/container/user"
 
[storage.options.overlay]
ignore_chown_errors = "true"
 
[storage.options]
additionalimagestores = ['/tmp1/imagestore']
 
 
#(Background for this https://github.com/containers/podman/issues/16608)
[user@workstation /tmp1]$ grep user /etc/su*id
[user@workstation /tmp1]$ 
  
 
[user@workstation /tmp1]$ mkdir imagestore
[user@workstation /tmp1]$ chmod 777 imagestore
[user@workstation /tmp1]$ mkdir mycontainers
[user@workstation /tmp1]$ ls -l /tmp1/
total 3673888
drwxrwxrwx  4 user extuser         36 Jul 26 13:26 container
-rwxr-xr-x  1 user extuser       4927 Jul 24 15:34 container_isvcode2
-rwxr-xr-x  1 user extuser        950 Jul 24 16:08 create_localconfig
drwxrwxrwt  3 root    root         4096 Jul 26 13:28 flowinfo
drwxrwxrwx  2 user extuser          6 Jul 26 16:38 imagestore
drwxrwxrwt 12 root    root          239 Jul 26 13:28 lsf
drwxr-xr-x  2 user extuser          6 Jul 26 16:38 mycontainers
-rw-rw-r--  1 user extuser 3762041344 Jul 24 16:06 isvcode_customerhpc-2021.02.tar
 
```

Actual issue:
```
[user@workstation /tmp1]$ podman --root /tmp1/mycontainers system migrate
[user@workstation /tmp1]$ podman --root /tmp1/mycontainers load -i /tmp1/isvcode_customerhpc-2021.02.tar
ERRO[0000] cannot find UID/GID for user user: no subuid ranges found for user "user" in /etc/subuid - check rootless mode in man pages. 
WARN[0000] Using rootless single mapping into the namespace. This might break some images. Check /etc/subuid and /etc/subgid for adding sub*ids if not using a network user 
Getting image source signatures
Copying blob f13b9edf1f31 done  
Copying blob e6e152cad445 done  
Copying blob d59071c74316 done  
Copying blob e4b4224f77ad done  
Copying blob fe18d8308335 done  
Error: payload does not match any of the supported image formats:
* oci: initializing source oci:/tmp1/isvcode_customerhpc-2021.02.tar:: open /tmp1/isvcode_customerhpc-2021.02.tar/index.json: not a directory
* oci-archive: loading index: open /var/tmp/oci242845501/index.json: no such file or directory
* docker-archive: writing blob: adding layer with blob "sha256:e6e152cad445e1bdb3bc77a0e55d378c65a150294a1a0f217533ef3d7ffa4fca": processing tar file(potentially insufficient UIDs or GIDs available in user namespace (requested 0:5 for /usr/bin/write): Check /etc/subuid and /etc/subgid if configured locally and run podman-system-migrate: lchown /usr/bin/write: invalid argument): exit status 1
* dir: open /tmp1/isvcode_customerhpc-2021.02.tar/manifest.json: not a directory
```

### Steps to reproduce the issue

See example above.

@vrothberg Was able to easily reproduce the error with a more current Fedora setup.

> It stops working once `--root` is used

### Describe the results you received

Image not being loaded to the additional image store

### Describe the results you expected

Image being loaded to the additional image store

### podman info output

```yaml
[user@workstation ~]$ cat /etc/redhat-release
Red Hat Enterprise Linux release 8.7 (Ootpa)
[user@workstation ~]$ podman -v
podman version 4.2.0
[user@workstation ~]$ crun --version
crun version 1.5
commit: 54ebb8ca8bf7e6ddae2eb919f5b82d1d96863dea
spec: 1.0.0
+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
```
```


### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by vrothberg** [comment]
Thanks for opening the issue, @holgrrr !

@giuseppe @rhatdan,  @holgrrr and I looked at this issue yesterday and I could reproduce:
* Drop entries from /etc/sub*id
* Configure storage.conf to ignore chown errors

This works just fine but if I add `--root $default_path` Podman fails.  I did not track it down yet but my expectation is that it should just work.

**Comment by giuseppe** [comment]
I think we reset the options when using a different root because the settings from one store might not apply to another.

You can add the option back with `--storage-opt "overlay.ignore_chown_errors=true"`:

```
podman --root /tmp1/mycontainers --storage-opt "overlay.ignore_chown_errors=true" load -i /tmp1/isvcode_customerhpc-2021.02.tar
```

**Comment by giuseppe** [comment]
for the record, it is overwritten here: https://github.com/containers/podman/blob/main/libpod/options.go#L82

**Comment by vrothberg** [comment]
Thanks for taking a look, @giuseppe.

In the context of this issue, the behavior of dropping storage.conf options is surprising but I don´t we can change it due to the risk of breaking other users.

I guess we need a flag to instruct Podman to not drop the storage.conf options. WDYT?

**Comment by giuseppe** [comment]
not dropping options might have side effects (e.g., overlay redirect or mount_program) that change how layers are stored.  Not sure how to make clear for the user that the settings are ignored though.

Since the user is already expected to know about this behavior for using the new option, wouldn't it be simpler just to specify the ones to use with `--storage-opt`?

**Comment by rhatdan** [comment]
I would go with the --storage-opt option.
